packages/slate-hyperscript/src/creators.ts

```diff
@@ -152,13 +152,13 @@ export function createSelection(
   const anchor: AnchorToken = children.find(c => c instanceof AnchorToken)
   const focus: FocusToken = children.find(c => c instanceof FocusToken)
 
-  if (!anchor || !anchor.offset || !anchor.path) {
+  if (!anchor || (!anchor.offset && anchor.offset !== 0) || !anchor.path) {
```

Comment:
Could this be written like this, to make it a little more readable?

`if (!anchor || !Number.isInteger(anchor.offset) || !anchor.path)`

Reply:
I would also prefer that, but as I mentioned in the description `Number.isFinite` (and `Number.isInteger`) doesn't accept a `number | undefined` union.

Reply:
```suggestion
  if (!anchor || anchor.offset === undefined || anchor.path === undefined) {
```

Reply:
Thanks for the suggestion, @sgurenkov! What I don't like about explicitly checking equality with `undefined` is that it doesn't include `null`. Which is fine in this case as long as you're in TypeScript-land, but not so much for JS. However, the chances for a `null` value slipping through to this point seem pretty slim.

Personally, I don't really care, though. Since you are both Slate contributors I'll leave the decision to you. :)

Reply:
`anchor.offset == null` is probably better. It will return false for `0` and true for `null` and `undefined`.

Reply:
Neat! Fixed in 4406a26. Thanks!